File: server/Model/Entities/Organization/DepartmentType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Model.Entities.System;

// ...existing using statements...

namespace Model.Entities.Organization;

[Table("DepartmentTypes")]
public class DepartmentType : BaseEntity<int>
{
    [Required] [StringLength(50)] public string Code { get; set; } = string.Empty;

    [Required] [StringLength(100)] public string Name { get; set; } = string.Empty;

    [StringLength(1000)] public string? Description { get; set; }

    public int Level { get; set; }

}

File: client/types/sys/DepartmentType.ts
export type DepartmentTypeDto = {
  id: number;
  code: string;
  name: string;
  description?: string;
  level?: number;
};

export const defaultDepartmentTypeDto: DepartmentTypeDto = {
  id: 0,
  code: '',
  name: '',
  description: '',
  level: 0,
};

export type SaveDepartmentTypeDto = DepartmentTypeDto;
